domainservice/src/main/kotlin/systems/ajax/malov/stockanalyzer/mapper/proto/GetFiveBestStockSymbolsWithStockRecordsRequestMapper.kt

```diff
@@ -0,0 +1,60 @@
+package systems.ajax.malov.stockanalyzer.mapper.proto
+
+import com.google.protobuf.ByteString
+import systems.ajax.malov.input.reqreply.stock.get_n_best_stock_symbols_with_stocks.proto.AggregatedStockRecordItemResponse
+import systems.ajax.malov.input.reqreply.stock.get_n_best_stock_symbols_with_stocks.proto.GetNBestStockSymbolsWithStockRecordsResponse
+import systems.ajax.malov.internalapi.commonmodel.stock.big_decimal.proto.BDecimal
+import systems.ajax.malov.internalapi.commonmodel.stock.big_decimal.proto.BInteger
+import systems.ajax.malov.internalapi.commonmodel.stock.short_stock.proto.ShortStockRecordResponse
+import systems.ajax.malov.stockanalyzer.entity.MongoStockRecord
+import java.math.BigDecimal
+import java.math.BigInteger
+
+object GetFiveBestStockSymbolsWithStockRecordsRequestMapper {
+    fun toGetFiveBestStockSymbolsWithStockRecordsRequest(aggregatedData: Map<String, List<MongoStockRecord>>):
+        GetNBestStockSymbolsWithStockRecordsResponse {
+        return GetNBestStockSymbolsWithStockRecordsResponse.newBuilder().apply {
+            successBuilder.addAllStockSymbols(
+                aggregatedData.map { (symbol, stocks) ->
+                    toAggregatedStockRecordItemResponseDto(symbol, stocks)
+                }
+            )
+        }.build()
```

Comment:
let's refactor this code for practice purpose.

Use `fold` or `reduce` function